integration-tests/soak/vrfv2_test.go

```diff
@@ -0,0 +1,363 @@
+	},
+		chainlinkNodes)
+
+	t.Cleanup(func() {
+		if err := actions.TeardownRemoteSuite(vrfV2SoakTest.TearDownVals(t)); err != nil {
+			l.Error().Err(err).Msg("Error tearing down environment")
+		}
+	})
+	vrfV2SoakTest.Setup(t, testEnvironment)
+	l.Info().Msg("Set up soak test")
+	vrfV2SoakTest.Run(t)
+}
+
+func setupVRFV2Environment(
+	t *testing.T,
+	testNetwork blockchain.EVMNetwork,
+	networkDetailTomlConfig string,
+	existingNamespace string,
+	newEnvLabel string,
+) (testEnvironment *environment.Environment) {
+	gethChartConfig := getGethChartConfig(testNetwork)
+
+	if existingNamespace != "" {
+		testEnvironment = environment.New(&environment.Config{
+			Namespace: existingNamespace,
+			Test:      t,
+			TTL:       time.Hour * 1,
+			Labels:    []string{newEnvLabel},
+		})
+	} else {
+		testEnvironment = environment.New(&environment.Config{
+			NamespacePrefix: fmt.Sprintf("soak-vrfv2-%s", strings.ReplaceAll(strings.ToLower(testNetwork.Name), " ", "-")),
+			Test:            t,
+			TTL:             time.Hour * 1,
+		})
+	}
+
+	testEnvironment = testEnvironment.
+		AddHelm(gethChartConfig).
+		AddHelm(chainlink.New(0, map[string]any{
+			"toml": client.AddNetworkDetailedConfig("", networkDetailTomlConfig, testNetwork),
+			//need to restart the node with updated eth key config
+			"db": map[string]interface{}{
+				"stateful": "true",
+			},
+		}))
+	err := testEnvironment.Run()
+	require.NoError(t, err, "Error running test environment")
+	return testEnvironment
+}
+
+func getGethChartConfig(testNetwork blockchain.EVMNetwork) environment.ConnectedChart {
+
+	evmConfig := ethereum.New(nil)
+	if !testNetwork.Simulated {
+		evmConfig = ethereum.New(&ethereum.Props{
+			NetworkName: testNetwork.Name,
+			Simulated:   testNetwork.Simulated,
+			WsURLs:      testNetwork.URLs,
+		})
+	}
+	return evmConfig
+}
+
+func WaitForRandRequestToBeFulfilled(
+	consumer contracts.VRFv2Consumer,
+	lastRequestID *big.Int,
+	timeout time.Duration,
+	wg *sync.WaitGroup,
+	t *testsetups.VRFV2SoakTest,
+) (contracts.RequestStatus, error) {
+	requestStatusChannel := make(chan contracts.RequestStatus)
+	requestStatusErrorsChannel := make(chan error)
+
+	// set the requests to only run for a certain amount of time
+	testContext, testCancel := context.WithTimeout(context.Background(), timeout)
+	defer testCancel()
+
+	ticker := time.NewTicker(time.Second * 1)
+
+	for {
+		select {
+		case <-testContext.Done():
+			ticker.Stop()
+			wg.Done()
+			return contracts.RequestStatus{}, fmt.Errorf("timeout waiting for new transmission event")
+
+		case <-ticker.C:
+			go getRandomnessRequestStatus(
+				consumer,
+				lastRequestID,
+				requestStatusChannel,
+				requestStatusErrorsChannel,
+			)
+
+		case requestStatus := <-requestStatusChannel:
+			if requestStatus.Fulfilled == true {
+				t.NumberOfFulfillments++
+				wg.Done()
+				return requestStatus, nil
+			}
+
+		case err := <-requestStatusErrorsChannel:
+			wg.Done()
+			return contracts.RequestStatus{}, err
+		}
```

Comment:
I think this can be simplified without using additional channels for errors and request status